src/wasm/wasm.cpp

```diff
@@ -381,13 +381,15 @@ void SetGlobal::finalize() {
 }
 
 void Load::finalize() {
+  assert(!isAtomic || type == i32 || type == i64 || type == unreachable);
```

Comment:
I think this and the assert below should be covered by the validator (not in an assert, but also in release builds there).

Reply:
Done.